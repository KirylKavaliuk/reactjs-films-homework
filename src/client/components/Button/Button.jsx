import React from 'react';

import styles from './Button.scss';

const Button = ({ label, onClick, transparent }) => {
  const transparentClass = transparent ? styles.transparent : '';

  return <button
    className={ `${styles.button} ${transparentClass}`}
    onClick={ onClick }
  >
    { label }
  </button>;
};

export default Button;
