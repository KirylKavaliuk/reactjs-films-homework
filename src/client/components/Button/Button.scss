.button {
  font-size: 16px;
  background: #09ade3;
  border: 1px solid #09ade3;
  padding: 10px;
  min-width: 130px;
  border-radius: 3px;

  &:focus {
    outline: none;
  }
}

.transparent {
  background: none;
  border: 1px solid #fff;
}
