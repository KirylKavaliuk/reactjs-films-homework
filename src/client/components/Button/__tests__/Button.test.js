import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Button from '../Button';

it('renders button shallow correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Button label='button'/>);

  const tree = renderer.getRenderOutput();

  expect(tree).toMatchSnapshot();
});
