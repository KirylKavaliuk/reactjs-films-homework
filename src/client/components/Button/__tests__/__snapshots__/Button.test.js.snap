// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button shallow correctly 1`] = `
<button
  className="Button__button___1nbn1"
  onClick={null}
>
  button
</button>
`;
